src/UXClient/Components/LineChart/LineChart.ts

```diff
@@ -559,253 +551,111 @@ class LineChart extends TemporalXAxisComponent {
-            var intervalCenter = Number(intervalCenterString);
-            if (Math.abs(intervalCenter - prevMillis) < minDistance) {
-                minDistance = Math.abs(intervalCenter - prevMillis);
-                closestValue = intervalCenter;
-            }
-        });
-        return closestValue;
+    public exportMarkers () {
+
+        this.chartOptions.markers = Object.keys(this.markerGuidMap).map((markerGuid) => this.markerGuidMap[markerGuid].getMillis());
+        this.chartOptions.onMarkersChange(this.chartOptions.markers);
     }
 
-    private setScooterPosition (scooter, rawMillis: number = null) {
-        if (!scooter) {
-            return;
-        }
-        var closestTime;
-        if (rawMillis != null) {
-            closestTime = this.findClosestValidTime(rawMillis);
-            this.scooterGuidMap[scooter.datum()] = closestTime;    
-        }
-        closestTime = this.scooterGuidMap[scooter.datum()];
-        if (closestTime < this.chartComponentData.fromMillis || closestTime > this.chartComponentData.toMillis) {
-            scooter.style("display", "none");
-        } else {
-            scooter.style("display", "block")
-                .style("left", (d) => {
-                    var closestTime = this.scooterGuidMap[d];
-                    return (Math.round(this.x(closestTime) + this.getScooterMarginLeft()) + "px");
-                })
-                .style("top", this.chartMargins.top + this.chartOptions.aggTopMargin + "px")
-                .style("height", this.height - (this.chartMargins.top + this.chartMargins.bottom + this.chartOptions.aggTopMargin) + "px");
+    private createOnMarkerChange (markerGuid: string, marker: any) {
+        return (isDeleting, droppedMarker) => {
+            if (droppedMarker) {
+                this.markerGuidMap[markerGuid] = marker;
+                this.isDroppingMarker = false;
+            }
+            else if (isDeleting) {
+                delete this.markerGuidMap[markerGuid];
+
+                //set focus to first marker if markers exist on delete
+                if (Object.keys(this.markerGuidMap).length !== 0) {
+                    this.markerGuidMap[Object.keys(this.markerGuidMap)[0]].focusCloseButton();
+                } else {
+                    this.focusOnEllipsis();
+                }
+
+            } 
+            this.exportMarkers();
+            this.sortMarkers();
         }
+    }
 
-        d3.select(this.renderTarget).selectAll(".tsi-scooterContainer").sort((a: string, b: string) =>  { 
-            return (this.scooterGuidMap[a] < this.scooterGuidMap[b]) ? 1 : -1;            
-        });
+    private renderMarker (marker: any, millis: number, onChange: any = null) {
+        marker.render(millis, this.chartOptions, this.chartMargins, this.chartComponentData, this.x, this.height, this.getMarkerMarginLeft(), this.colorMap, this.yMap, onChange, false);
```

Comment:
wondering if we can use fewer parameters, and instead use an object as a parameter, and use object destructuring or strong typing of the object parameter to name the fields.  that allows us to add new properties to the parameter object and destructure it, without having to add new parameters.  might just be a cleaner pattern